bowling/Game.cpp

```diff
@@ -1,203 +1,140 @@
 #include "Game.hpp"
+#include "Frame.hpp"
 
 #include <algorithm>
 #include <fstream>
 #include <iostream>
+#include <sstream>
+#include <stdexcept>
 
-Game::Game(const std::filesystem::path& filePath) {
+namespace fs = std::filesystem;
+
+Game::Game(const fs::path& filePath) {
     loadFromFile(filePath.string());
 }
 
 void Game::loadFromFile(const std::string& filePath) {
     std::ifstream file(filePath, std::ios_base::in);
     if (!file.good()) {
-        std::cerr << "file " << filePath << " could not be opened!\n";
-        abort();
+        throw std::invalid_argument("file " + filePath + " could not be opened!\n");
     }
     while (!file.eof()) {
-        PlayerData player;
-        std::getline(file, player.name, ':');
-        loadPlayerRolls(file, player.rolls);
+        std::string playerName{};
+        std::vector<Frame> playerRolls{};
+        std::getline(file, playerName, ':');
+        playerRolls = getPlayerRolls(file);
         if (!file.eof()) {
-            players.push_back(player);
+            if (std::find_if(players.begin(), players.end(), [&](auto& player) {
+                    return player.getName() == playerName && player.getRolls() == playerRolls;
+                }) == players.end()) {
+                players.emplace_back(PlayerData(playerName, playerRolls));
+            }
```

Comment:
can we have duplicates? Or what is the reason for using this find_if? I recommend using std::set / unordered_set if you want to handle duplicates automatically.

Reply:
I thought about using `loadFromFile()` function many times on a single file, that's why I decided to only add a player if it's not yet in the data base. But maybe it would be a better idea to clear the data base before loading players.